doc/modules/preprocessing.rst

```diff
@@ -480,6 +480,39 @@ values than observed values.
 :class:`Imputer` can be used in a Pipeline as a way to build a composite
 estimator that supports imputation. See :ref:`sphx_glr_auto_examples_missing_values.py`
 
+.. _missing_indicator:
+
+Transformer indicating missing values
+=====================================
+
+:class:`MissingIndicator` transformer is useful to transform a dataset into corresponding
+binary matrix indicating the presence of missing values in the dataset.
+The knowledge of which features were imputed can be exploited by a downstream
+estimator by adding features that indicate which elements have been imputed.
+
+    >>> from sklearn.preprocessing import MissingIndicator
+    >>> import numpy as np
+    >>> a = MissingIndicator(missing_values = -1)
+    >>> X1 = np.array([
+    ...   [-1, -1,  1,  3],
+    ...   [ 4, -1,  0, -1],
+    ...   [ 8, -1,  1,  0],
+    ... ])
+    >>> X2 = np.array([
+    ...   [ 5,  1, -1, -1],
+    ...   [-1, -1,  2,  3],
+    ...   [ 2,  3,  4,  0],
+    ... ])
+    >>> MI = MissingIndicator(missing_values = -1)
```

Comment:
Why do you make a new instance? Also `MI` is not an appropriate variable name for an estimator instance.